MetaMask/metamask-extension
# fix: Improve AccountListMenu/Item performance

## **Description**
1. What is the reason for the change?
- reduce unnecessary re renders of the AccountsListMenu and AccountsListItem component
- reduce tech debt
- adopt better patterns like memoizing callbacks, lifting state to parent components and memoizing smaller components
2. What is the improvement/solution?
- memoize AccountsListItem
- remove `getSelectedInternalAccount` form AccountsListItem since this component already has a prop called `selected` that uses the same logic. Lifting this state to the parent means that the `AccountsListItem` will not need to re render. Also a change in the `getSelectedInternalAccount` would trigger a re render of all its parents which is unnecessary. 
    - This prop is now required to make up for the fact that we don't have the state anymore
- memoize callbacks in the AccountListMenu

- _**This PR has minor changes and should not effect any user flows. It does not require QA before merging**_

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26379?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/512

## **Manual testing steps**
- import / create a wallet
- create as many accounts as you can
- each time you create an account, the new account should be selected
- open the account menu and select a new account, the account should become globally selected
- select an account, click the three dots beside it, go to account details, click on view private key, put password, the private key should be displayed

## **Screenshots/Recordings**



### **Before**



### **After**


https://github.com/user-attachments/assets/9e57fd80-88d1-4708-95e2-303e19106bed



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
